app/services/backup_code_generator.rb

```diff
@@ -0,0 +1,70 @@
+require 'digest'
+
+class BackupCodeGenerator
+  attr_reader :length
+
+  NUMBER_OF_CODES = 10
+
+  def initialize(user, length: 2, split: 4)
+    @length = length
+    @split = split
+    @user = user
+  end
+
+  def generate
+    delete_existing_codes
+    generate_new_codes
+  end
+
+  def create
+    @user.save
+    generate
+  end
+
+  def verify(plaintext_code)
+    backup_code = normalize(plaintext_code)
+    code = BackupCodeConfiguration.find_with_code(code: backup_code, user_id: @user.id)
+    return unless code_usable?(code)
+    code.update!(used: true, used_at: Time.zone.now)
+    true
+  end
+
+  def delete_existing_codes
+    @user.backup_code_configurations.destroy_all
+  end
+
+  def generate_new_codes
+    result = []
+    (0..(NUMBER_OF_CODES - 1)).each do
```

Comment:
Style: Or, since we don't need the index, `(1 .. NUMBER_OF_CODES).each do ... end` or even
```ruby
def generate_new_codes
  (1 .. NUMBER_OF_CODES).map do |_|
    code = backup_code
    save_code(code)
    code
  end
end
```

Reply:
done